blockstore/src/main/java/brooklyn/location/blockstore/vclouddirector15/VcloudVolumeManager.java

```diff
@@ -0,0 +1,119 @@
+    @Override
+    public BlockDevice createBlockDevice(JcloudsMachineLocation jcloudsMachineLocation, BlockDeviceOptions options) {
+        Optional<NodeMetadata> vcloudNodeMetadata = jcloudsMachineLocation.getOptionalNode();
+        VCloudDirectorApi vCloudDirectorApi = jcloudsMachineLocation.getParent().getComputeService().getContext().unwrapApi(VCloudDirectorApi.class);
+        VmApi vmApi = vCloudDirectorApi.getVmApi();
+        TaskApi taskApi = vCloudDirectorApi.getTaskApi();
+        Vm vm = Vm.builder().id(vcloudNodeMetadata.get().getId()).build();
+        RasdItemsList virtualHardwareSectionDisks = vmApi.getVirtualHardwareSectionDisks(vm.getId());
+
+        if (!Iterables.tryFind(virtualHardwareSectionDisks, NewScsiLogicSASDisk.SCSI_LSILOGICSAS_PREDICATE).isPresent()) {
+            virtualHardwareSectionDisks = new AddScsiLogicSASBus().addScsiLogicSASBus(virtualHardwareSectionDisks);
+        }
+
+        RasdItem nextDisk = new NewScsiLogicSASDisk().apply(virtualHardwareSectionDisks);
+
+        CimString newDiskHostResource = new CimString(Iterables.getOnlyElement(nextDisk.getHostResources()));
+        Preconditions.checkNotNull(newDiskHostResource, "HostResource for the existing disk should not be null");
+        newDiskHostResource.getOtherAttributes().put(new QName("http://www.vmware.com/vcloud/v1.5", "capacity"), "" + (64 * 1024));
+        RasdItem newDiskToBeCreated = RasdItem.builder()
+                .fromRasdItem(nextDisk) // The same AddressOnParent (SCSI Controller)
+                .hostResources(ImmutableList.of(newDiskHostResource)) // NB! Use hostResources to override hostResources from newDisk
+                .build();
+        virtualHardwareSectionDisks.add(newDiskToBeCreated);
+        Task task = vmApi.editVirtualHardwareSectionDisks(vm.getId(), virtualHardwareSectionDisks);
+        Predicates2.retry(
+                new TaskSuccess(taskApi),
+                Predicates2.DEFAULT_PERIOD * 5L,
+                Predicates2.DEFAULT_MAX_PERIOD * 5L,
+                EDIT_VM_TIMEOUT_MS).apply(task);
+
+        return new VcloudBlockDevice(newDiskToBeCreated, jcloudsMachineLocation, vm);
+    }
+
+
+    // TODO
+    @Override
+    public AttachedBlockDevice attachBlockDevice(JcloudsMachineLocation machine, BlockDevice blockDevice, BlockDeviceOptions options) {
+        return new Devices.AttachedBlockDeviceImpl(machine, null, "/dev/sdb");
+    }
+
+    @Override
+    public BlockDevice detachBlockDevice(AttachedBlockDevice attachedBlockDevice) {
+        return null;
+    }
+
+    @Override
+    public void deleteBlockDevice(BlockDevice blockDevice) {
+
+    }
+
+    protected Volume waitForVolumeToBeAvailable(final BlockDevice device) {
+        final AtomicReference<Volume> lastVolume = new AtomicReference<Volume>();
+
+        boolean available = Repeater.create("waiting for volume available:" + device)
+                .every(1, TimeUnit.SECONDS)
+                .limitTimeTo(60, TimeUnit.SECONDS)
```

Comment:
It doesn't take more than 20 seconds to do that.
If it is more then there is a problem.